### DISCUSSION TITLE
[ideas] `description` field for targets

### DISCUSSION THREAD

**Discussion by tjmahr** [ropensci/targets#1230] [Ideas]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Sometimes I wish I had an `description` field for documenting what each target it is a more human-friendly way. 

```
tar_target(
  name,
  command,
  description = "a field where we can briefly describe the target",
)
```

One could write a code comment, but this approach would be better because it could show up in `tar_manifest()`. It would also help generate READMEs.

Arguably, we should make the commands into simple functions and document those functions, but the trouble I have is having several end-products and recalling what they are.

**Comment by wlandau** [comment]
If your targets do not take up too much storage, then you can do pretty much the same thing with a `tar_render()`/`tar_quarto()` report  which shows and explains each target you want to describe.

I will think about the description feature you propose, since I think it could also extend to https://github.com/ropensci/targets/discussions/1136 and #1196

**Reply by wlandau** [reply]
implemented in #1236
